debug_toolbar/decorators.py

```diff
@@ -11,7 +11,7 @@ def require_show_toolbar(view):
     def inner(request, *args, **kwargs):
         from debug_toolbar.middleware import get_show_toolbar
 
-        show_toolbar = get_show_toolbar()
+        show_toolbar = get_show_toolbar(async_mode=False)
```

Comment:
Shouldn't we pass `async_mode` based on the type of request `WSGIRequest` or `ASGIRequest` here ? I mean I am not sure whether `require_show_toolbar` would ever be exposed to a custom `async view` in async context or vice versa but this would save the conversion as it slows down  the overall process. thoughts?

Reply:
I was hoping `require_show_toolbar` was an undocumented API, but it is documented. Yes, we need to pass in a better value than always `False`. Good catch @salty-ivy!

It feels like we'd want `async_mode` defined by whether `view` is a coroutine though rather than based on WSGIRequest vs ASGIRequest. Not 100% sure though.

Reply:
It can possible break in a situation where everything is in async context
including `show_toolbar` call back and if I use `require_show_toolbar` on an `async view` it will forcefully convert it into a sync func and we would get into similar error given that `require_show_toolbar` is exposed to such conditions :D

Reply:
> It feels like we'd want `async_mode` defined by whether `view` is a coroutine though rather than based on WSGIRequest vs ASGIRequest. Not 100% sure though.

may be checking both but it seems like that would cause a rabbit hole : (

Reply:
Oh. I totally missed that this decorator needs to be async friendly.

Reply:
> It feels like we'd want `async_mode` defined by whether `view` is a coroutine though rather than based on WSGIRequest vs ASGIRequest. Not 100% sure though.

This is correct, we should check whether view is a coroutine or not since django also converts the request object based on that, just checked : D

Reply:
yeah missed that too and yeah a iscoroutinefunction is the way other django decorator work.